### PR TITLE
Feat: Module Loader [load(path, options)] method

### PR DESCRIPTION
**Pull request by EternalQuasar0206**
Usage:

- Basic import:
```js
const variables = load(__basedir + "/vars.js")
```

- Using options:
```js
const variables = load(__basedir + "/vars.js", {
    useUnsafeInjectorLoader: false, //[Not recommended] Loads and executes the script directly
    useHttpRequest: false //Uses fs to load the files
})
```

- Using **unsafe injector loader** (not recommended):
```js
load(__basedir + "/vars.js", {
    useUnsafeInjectorLoader: true
})
```

### CODE REVIEW

File: MelonJs.Experiments/FileCrudApi/index.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
-load(__basedir + "/routes/getEntriesRoute.js")
-load(__basedir + "/routes/writeEntryRoute.js")
-load(__basedir + "/routes/deleteEntryRoute.js")
+const loadOptions = {
+    useUnsafeInjectorLoader: true,
+    useHttpRequest: false
+}
```
Comment (NicolasLopes7) [inline MelonJs.Experiments/FileCrudApi/index.js:4]: shouldn't it be abstracted on another layer?

File: MelonJs.JavaScript/Bindings/Tools/load.js [hunk 2]
```diff
@@ -1,4 +1,85 @@
-﻿const load = (path) => {
-    const content = fs.read(path);
-    melon_internal_script_injector(content);
+﻿const load = (path, options = { useHttpRequest: true, useUnsafeInjectorLoader: false }) => {
```
Comment (NicolasLopes7) [inline MelonJs.JavaScript/Bindings/Tools/load.js:1]: why `useHttpRequest ` is set as true by default?

File: MelonJs.JavaScript/Bindings/Tools/load.js [hunk 3]
```diff
@@ -1,4 +1,85 @@
...
+    const content = options.useHttpRequest ? fs.read(path) : http.request(path);
```
Comment (NicolasLopes7) [inline MelonJs.JavaScript/Bindings/Tools/load.js:2]: shouldn't be?
```suggestion
    const content = options.useHttpRequest ? http.request(path) : fs.read(path);
```

File: MelonJs.JavaScript/Bindings/Tools/load.js [hunk 4]
```diff
@@ -1,4 +1,85 @@
...
+                value = new ConstructorAssembler(input.init.callee.name, input.init.arguments);
+                break;
+
+            //TODO: ADD SUPPORT TO ArrowFunctionExpression AND FunctionExpression
+        }
+
+        return {
+            name,
+            value
+        };
```
Comment (NicolasLopes7) [inline MelonJs.JavaScript/Bindings/Tools/load.js:59]: move it to an external function

File: MelonJs.JavaScript/Bindings/Tools/load.js [hunk 5]
```diff
@@ -1,4 +1,85 @@
...
+            return {
+                success: true
+            }
+        }
+        catch (e) {
+            return {
+                success: false,
+                error: e.toString()
+            }
+        }
```
Comment (NicolasLopes7) [inline MelonJs.JavaScript/Bindings/Tools/load.js:17]: move it to a function

File: MelonJs.JavaScript/Bindings/Tools/load.js [hunk 6]
```diff
@@ -1,4 +1,85 @@
...
+                    value[property.key.name] = getDeclarationPatternValue(property.value.value);
+                });
+                break;
+
+            case "NewExpression":
+                value = new ConstructorAssembler(input.init.callee.name, input.init.arguments);
+                break;
+
+            //TODO: ADD SUPPORT TO ArrowFunctionExpression AND FunctionExpression
+        }
```
Comment (NicolasLopes7) [inline MelonJs.JavaScript/Bindings/Tools/load.js:54]:
```ts
const getValueByTokenType = {
 "Literal": () => {...},
"ArrayExpression": () => {...},
"ObjectExpression": () => {...},
"NewExpression": () => {...},
}

const value = getValueByTokenType[input.init.type]()

return { name, value }

Comment (EternalQuasar0206) [reply MelonJs.JavaScript/Bindings/Tools/load.js:54]: Using my own spell against me?

File: MelonJs.JavaScript/Bindings/Tools/load.js [hunk 7]
```diff
@@ -1,4 +1,85 @@
...
+            name,
+            value
+        };
+    }
+
+    parsed.forEach(item => { 
+        item.declarations.forEach(declaration => {
+            result.push(getDeclarationPatternValue(declaration));
+        });
+    });
```
Comment (NicolasLopes7) [inline MelonJs.JavaScript/Bindings/Tools/load.js:66]:
```suggestion
const result = parsed
  .flatMap((item) => item.declarations)
  .map(getDeclarationPatternValue)

File: MelonJs.JavaScript/Bindings/Tools/load.js [hunk 8]
```diff
@@ -1,4 +1,85 @@
...
+            return {
+                success: false,
+                error: e.toString()
+            }
+        }
+    }
+
+    const parsed = esprima.parse(content).body;
+    const result = [];
+    const declared = {};
```
Comment (NicolasLopes7) [inline MelonJs.JavaScript/Bindings/Tools/load.js:22]: move the variable closer than where it being used

**Review by NicolasLopes7 — APPROVED** [review]
Left some comments, but LGTM!

File: MelonJs.JavaScript/Bindings/Tools/load.js (hunk 6 above)
Comment (NicolasLopes7) [reply MelonJs.JavaScript/Bindings/Tools/load.js:54]: yup 🧙 🪄

File: MelonJs.Experiments/FileCrudApi/index.js (hunk 1 above)
Comment (EternalQuasar0206) [reply MelonJs.Experiments/FileCrudApi/index.js:4]: How?

File: MelonJs.JavaScript/Bindings/Tools/load.js (hunk 2 above)
Comment (EternalQuasar0206) [reply MelonJs.JavaScript/Bindings/Tools/load.js:1]: It depends when it is being used, but fs would be cool too

File: MelonJs.JavaScript/Bindings/Tools/load.js (hunk 3 above)
Comment (EternalQuasar0206) [reply MelonJs.JavaScript/Bindings/Tools/load.js:2]: yes

File: MelonJs.JavaScript/Bindings/Tools/load.js (hunk 6 above)
Comment (EternalQuasar0206) [reply MelonJs.JavaScript/Bindings/Tools/load.js:54]: done

File: MelonJs.JavaScript/Bindings/Tools/load.js (hunk 4 above)
Comment (EternalQuasar0206) [reply MelonJs.JavaScript/Bindings/Tools/load.js:59]: done i guess

**Pull request merged**